### PR TITLE
doc: update Overview with DX recommendations

### PR DESCRIPTION
**Pull request by amyreye**
- Capitalize Board Inspector and ACRN Configurator names to highlight them in images and content
- Remove duplicate links between the Overview and GSG where not needed
- Provide additional details that help to deepen developer’s knowledge where applicable.  Potential items: acrn-dm, more clarity of the relationship between the ServiceVM and UserVMs, expanded definitions of scenarios (beyond one liners).

Signed-off-by: Amy Reyes <amy.reyes@intel.com>

### CODE REVIEW

File: doc/getting-started/overview_dev.rst [hunk 1]
```diff
@@ -3,19 +3,22 @@
 Configuration and Development Overview
 ######################################
 
-This overview is for developers who are new or relatively new to ACRN. It will
-help you get familiar with ACRN basics: ACRN components and general process for
-building an ACRN hypervisor.
+This overview is for developers who are new or relatively new to ACRN and are
+responsible for configuring and building ACRN hypervisors. It will introduce you
```
Comment (dbkinder) [inline doc/getting-started/overview_dev.rst:7]: Technically, we're configuring and building the hypervisor and the VMs for the application.

File: doc/getting-started/overview_dev.rst [hunk 2]
```diff
@@ -3,19 +3,22 @@
 Configuration and Development Overview
 ######################################
 
-This overview is for developers who are new or relatively new to ACRN. It will
-help you get familiar with ACRN basics: ACRN components and general process for
-building an ACRN hypervisor.
+This overview is for developers who are new or relatively new to ACRN and are
```
Comment (dbkinder) [inline doc/getting-started/overview_dev.rst:6]: How about deleting "or relatively new"...

File: doc/getting-started/overview_dev.rst [hunk 3]
```diff
@@ -3,19 +3,22 @@
...
 The overview covers the process at an abstract and universal level.
 
 * Abstract: the overall structure rather than detailed instructions
 * Universal: applicable to most use cases
 
-Although the overview describes the process as a series of steps, it's intended
-to be a summary, not a step-by-step guide. Throughout the overview, you will see
-links to the :ref:`gsg` for first-time setup instructions. Links to advanced
-guides and additional information are also provided.
+The overview is intended to complement the :ref:`gsg`. The guide provides
```
Comment (dbkinder) [inline doc/getting-started/overview_dev.rst:15]:
```suggestion
This overview complements the :ref:`gsg`. The guide provides
```

File: doc/getting-started/overview_dev.rst [hunk 4]
```diff
@@ -3,19 +3,22 @@
...
+step-by-step instructions to enable an ACRN example for first-time use, while
+the overview provides background information and serves as a gateway to
+additional features and resources that can help you develop your solution.
+
+The overview doesn't cover ACRN benefits, use cases, or architecture. See
```
Comment (dbkinder) [inline doc/getting-started/overview_dev.rst:20]: How about a positive statement instead (drop the first sentence, "The overview doesn't..."):
```
See :ref:`introduction` for information about ACRN benefits, use cases, and architecture.
```

File: doc/getting-started/overview_dev.rst [hunk 5]
```diff
@@ -138,16 +150,13 @@ case, as described later in :ref:`overview_dev_config_editor`.
...
 * Build tools
 * ACRN hypervisor source code
 * If your scenario has a Service VM: ACRN kernel source code
 
-The :ref:`gsg` provides step-by-step instructions for setting up your
-development computer.
-
-In the next step, :ref:`overview_dev_board_config`, you will need the board
-inspector tool found in the ACRN hypervisor source code to collect information
+In the next step, :ref:`overview_dev_board_config`, you will need the Board
```
Comment (dbkinder) [inline doc/getting-started/overview_dev.rst:158]: Maybe move this sentence into the first paragraph of the next step.

File: doc/getting-started/overview_dev.rst [hunk 6]
```diff
@@ -158,101 +167,123 @@ about the target hardware and generate a board configuration file.
...
 
-You must configure all of your target's BIOS settings before running the board
-inspector tool, because the tool records the current BIOS settings in the board
+You must configure all of your target's BIOS settings before running the Board
+Inspector tool, because the tool records the current BIOS settings in the board
 configuration file.
 
-Some BIOS settings are required by ACRN. The :ref:`gsg` provides a list of the
-settings.
+ACRN requires the following BIOS settings:
```
Comment (dbkinder) [inline doc/getting-started/overview_dev.rst:190]: There's also a third BIOS setting (disable secure boot).  I'm not sure you want to duplicate these settings from the GSG here anyway (means we'd need to keep two places up to date).

File: doc/getting-started/overview_dev.rst [hunk 7]
```diff
@@ -35,8 +38,9 @@ for :ref:`debugging and system messaging <acrn-debug>`. If your target doesn't
 have a serial output, :ref:`here are some tips for connecting a serial output
 <connect_serial_port>`.
 
-You will need a way to copy the built ACRN images from the development computer
-to the target system. A USB drive is recommended.
+You need a way to copy the built ACRN images and other files between the
+development computer and target system. ACRN documentation, such as the Getting
```
Comment (gvancuts) [inline doc/getting-started/overview_dev.rst:42]: Should we link again here to the GSG?

File: doc/getting-started/overview_dev.rst [hunk 8]
```diff
@@ -81,18 +84,26 @@ that ACRN offers:
...
-  ACRN supports one Service VM.
-
-* **Post-launched User VMs**: These VMs share hardware resources. Unlike
-  pre-launched VMs, you can change the configuration at run-time. They are
+* **Pre-launched User VMs**: These VMs are automatically launched at boot time
+  by the hypervisor. They run independently of other VMs and own dedicated
+  hardware resources, such as a CPU core, memory, and I/O devices. Other VMs,
+  including the Service VM, may not even be aware of a pre-launched VM’s
+  existence. The configuration of pre-launched VMs is static and must be defined
+  at build time. They are well-suited for safety-critical applications.
```
Comment (gvancuts) [inline doc/getting-started/overview_dev.rst:92]: Can we add a note along those lines: `and where very strict isolation, including from the Service VM is desirable`?

File: doc/getting-started/overview_dev.rst [hunk 9]
```diff
@@ -282,8 +310,11 @@ then boot ACRN.
 
 At a high level, you will:
 
-* Copy the built ACRN hypervisor files, kernel files, and launch scripts from
-  the development computer to the target.
+* Copy the built ACRN hypervisor files, Service VM kernel files, and launch
+  scripts from the development computer to the target. The Service VM kernel
+  files replace parts of the Ubuntu installation we installed and used for
+  running the Board Inspector, with the Linux kernel we built based on the board
```
Comment (gvancuts) [inline doc/getting-started/overview_dev.rst:316]: I think this is a bit misleading to the extent there is nothing in the Linux kernel we provide that is taken from either the board or scenario XML.

File: doc/getting-started/overview_dev.rst [hunk 10]
```diff
@@ -265,12 +296,9 @@ typically takes a few minutes.
...
 file and a makefile to build the ACRN kernel binary and associated components.
-The build can take 1-3 hours depending on the performance of your development
-computer and network.
-
-The :ref:`gsg` provides step-by-step instructions.
-
-For more information about the kernel, see :ref:`kernel-parameters`.
+The kernel build can take 15 minutes or less on a fast computer, but could take
+an hour or more depending on the performance of your development computer. For
+more information about the kernel, see :ref:`kernel-parameters`.
```
Comment (gvancuts) [inline doc/getting-started/overview_dev.rst:301]:
```suggestion
more information about the kernel parameters, see :ref:`kernel-parameters`.
```

File: doc/getting-started/overview_dev.rst [hunk 11]
```diff
@@ -292,18 +323,12 @@ At a high level, you will:
...
 **********
 
 * To get ACRN up and running for the first time, see the :ref:`gsg` for
   step-by-step instructions.
 
-* If you have already completed the :ref:`gsg`, see the :ref:`develop_acrn` for
-  more information about complex scenarios, advanced features, and debugging.
+* If you have already completed the Getting Started Guide, see the
+  :ref:`develop_acrn` for more information about complex scenarios, advanced
+  features, and debugging
```
Comment (gvancuts) [inline doc/getting-started/overview_dev.rst:334]:
```suggestion
  features, and debugging.
```

File: doc/getting-started/overview_dev.rst [hunk 12]
```diff
@@ -292,18 +323,12 @@ At a high level, you will:
...
-Next Steps
+Learn More
 **********
 
 * To get ACRN up and running for the first time, see the :ref:`gsg` for
   step-by-step instructions.
 
-* If you have already completed the :ref:`gsg`, see the :ref:`develop_acrn` for
-  more information about complex scenarios, advanced features, and debugging.
+* If you have already completed the Getting Started Guide, see the
```
Comment (gvancuts) [inline doc/getting-started/overview_dev.rst:332]: Should we link to the GSG here?

File: doc/getting-started/overview_dev.rst (hunk 9 above)
Comment (dbkinder) [reply doc/getting-started/overview_dev.rst:316]: @gvancuts I thought we're making some tweaks to the linux kernel based on our configuration.  Is the make of the acrn-kernel not dependent on any of the scenario or hypervisor configuration options?

Comment (amyreye) [reply doc/getting-started/overview_dev.rst]: @dbkinder Ok, I removed it in the latest commit.

**Review by dbkinder — APPROVED** [review]
LGTM.  The CI failures are because of a networking issue, so this looks good to merge.

**Pull request merged**